packages/babel-plugin-transform-spread/src/index.js

```diff
@@ -5,13 +5,18 @@ import { types as t } from "@babel/core";
 export default declare((api, options) => {
   api.assertVersion(7);
 
-  const { loose, allowArrayLike } = options;
+  const iterableIsArray = api.assumption("iterableIsArray") ?? options.loose;
+  const arrayLikeIsIterable =
+    options.allowArrayLike ?? api.assumption("arrayLikeIsIterable");
```

Comment:
Ditto backwards.

Reply:
The reason for swapping the order here is that, while `loose` is generic (and thus it's good to keep it as a fallback value), this option was explicitly designed for this assumption and it has a quite specific name. Thus I followed the "locality principle": since they have the same specificity, the option closer to the plugin has the precedence.

I wasn't 100% sure of doing this, so I'm open to change it to match `loose`'s behavior if no one else thinks that `options.allowArrayLike` should take precedence.

Reply:
It makes sense to me that when `options.allowArrayLike` is specified, that it should take precedence over `api.assumption("arrayLikeIsIterable")`.